librepo/handle.c

```diff
@@ -841,7 +870,56 @@ lr_handle_setopt(LrHandle *handle,
     return ret;
 }
 
-static gboolean
+gboolean
+lr_handle_network_wait(LrHandle *handle, GError **err, guint seconds)
+{
+
+    assert(!err || *err == NULL);
+
+    if (!handle) {
+        g_set_error(err, LR_HANDLE_ERROR, LRE_BADFUNCARG,
+                    "No handle specified");
+        return FALSE;
+    }
+
+    GNetworkMonitor *monitor = g_network_monitor_get_default();
+    GCancellable *cancellable = g_cancellable_new ();
```

Comment:
What would be better here is to have this cancellable be a parameter to the API function.  As of right now, the cancellable here is doing nothing because nothing will ever cancel it.  If we make it part of the API, then e.g. the calling code can choose to cancel the wait.